### PR TITLE
The Types section in Core-Spec doc is out-dated

### PR DESCRIPTION
**Pull request by xnning**
In core-spec.pdf, Section 2.5 Types is out-dated:

- Functions have their own types `FunTy` now. It is no longer a specialized `ForAllTy` in Core. Though conceptually it is.
- The binder in `ForAllTy` has changed from `Binder` to `TyVarBinder`, whose `ArgFlag` tracks visibility of an argument.
- It is `TyBinder` now that tracks the difference between `Named` and `Anon` binders.

### CODE REVIEW

**Comment by nomeata** [comment]
I hope @goldfirere sees this.

File: docs/core-spec/.gitignore [hunk 1]
```diff
@@ -4,3 +4,4 @@
 CoreOtt.tex
 core-spec.tex
 *.fls
+**/auto
```
Comment (goldfirere) [inline docs/core-spec/.gitignore]: Why add this?

File: docs/core-spec/CoreSyn.ott [hunk 2]
```diff
@@ -113,8 +113,8 @@ t {{ tex \tau }}, k {{ tex \kappa }}, s {{ tex \sigma }}, phi {{ tex \phi }}
   | n                       ::   :: TyVarTy       {{ com \ctor{TyVarTy}: Variable }}
   | t1 t2                   ::   :: AppTy         {{ com \ctor{AppTy}: Application }}
   | T </ ti // i />         ::   :: TyConApp      {{ com \ctor{TyConApp}: Application of type constructor }}
-  | t1 -> t2                ::   :: FunTy         {{ com \ctor{ForAllTy (Anon ...) ...}: Function }}
-  | forall n . t            ::   :: ForAllTy      {{ com \ctor{ForAllTy (Named ...) ...}: Type and coercion polymorphism }}
+  | t1 -> t2                ::   :: FunTy         {{ com \ctor{FunTy}: Function }}
+  | forall n . t            ::   :: ForAllTy      {{ com \ctor{ForAllTy (TyVarBinder ...) ...}: Type and coercion polymorphism }}
```
Comment (goldfirere) [inline docs/core-spec/CoreSyn.ott]: You can leave out the `TyVarBinder` bit here.

File: docs/core-spec/core-spec.mng [hunk 3]
```diff
@@ -169,9 +169,10 @@ A program is just a list of bindings:
 
 \gram{\ottt}
 
-\ctor{ForAllTy}s are represented in two different ways, depending on whether
-the \ctor{ForAllTy} is anonymous (written $[[t1 -> t2]]$) or
-named (written $[[forall n . t]]$).
+\ctor{FunTy} is the special case for non-dependent function type. The
+\ctor{TyBinder} in \ghcfile{types/TyCoRep.lhs} distinguishes whether a binder is
+anonymours (\ctor{FunTy}) or named (\ctor{ForAllTy}). See
```
Comment (goldfirere) [inline docs/core-spec/core-spec.mng]: s/anonymours/anonymous/

File: docs/core-spec/core-spec.mng [hunk 4]
```diff
@@ -169,9 +169,10 @@ A program is just a list of bindings:
 
 \gram{\ottt}
 
-\ctor{ForAllTy}s are represented in two different ways, depending on whether
-the \ctor{ForAllTy} is anonymous (written $[[t1 -> t2]]$) or
-named (written $[[forall n . t]]$).
+\ctor{FunTy} is the special case for non-dependent function type. The
+\ctor{TyBinder} in \ghcfile{types/TyCoRep.lhs} distinguishes whether a binder is
```
Comment (goldfirere) [inline docs/core-spec/core-spec.mng]: But `TyBinder`s handle only non-anonymous binders. I think this needs a bit more revision.

Comment (xnning) [reply docs/core-spec/core-spec.mng]: Why is that? I thought the `Anon` constructor of `TyBinder` deals with anonymous binders.

File: docs/core-spec/.gitignore (hunk 1 above)
Comment (xnning) [reply docs/core-spec/.gitignore]: Removed.

File: docs/core-spec/CoreSyn.ott (hunk 2 above)
Comment (xnning) [reply docs/core-spec/CoreSyn.ott]: Done.

File: docs/core-spec/core-spec.mng (hunk 3 above)
Comment (xnning) [reply docs/core-spec/core-spec.mng]: Done.

File: docs/core-spec/core-spec.mng (hunk 4 above)
Comment (goldfirere) [reply docs/core-spec/core-spec.mng]: Oops -- sorry. I was thinking `TyVarBinder`.

File: docs/core-spec/core-spec.mng (hunk 3 above)
Comment (goldfirere) [reply docs/core-spec/core-spec.mng]: I don't see the change...

Comment (xnning) [reply docs/core-spec/core-spec.mng]: Oh my bad. Should be fixed now..

**Comment by bgamari** [comment]
Yes, I'm on it.

**Comment by bgamari** [comment]
Merged in 65c186f0fdde95fd7c63ab9bd9b33a0213dba7d1.

**Pull request closed without merging**